modules/cloudwatch_metrics/main.tf

```diff
@@ -0,0 +1,58 @@
+locals {
+  role_name     = regex(".*role/(?P<role_name>.*)$", var.lambda.lambda_function.role)["role_name"]
```

Comment:
I think I've seen this elsewhere, but technically this is not correct. A role has a required `name` and an optional `path` and both go into the ARN. 

https://stackoverflow.com/a/69002757

Reply:
Excellent catch. This was indeed cribbed from the snapshot module.

I will replace with equivalent of:
```

data "aws_arn" "role" {
  arn = var.lambda.lambda_function.role
}

local {
  role_name = trimprefix(data.aws_arn.role.resource, "role/")
}